src/sentry/issues/ingest.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import annotations
+
+from typing import TypedDict, Optional, Sequence, Mapping, Any
+
+from django.db import transaction
+from sentry import metrics
+from sentry.event_manager import EventManager
+from sentry.eventstore.models import Event
+from sentry.issues.issue_occurrence import IssueOccurrence
+from sentry.models import GroupHash
+
+
+class IssueEvidenceData(TypedDict):
+    name: str
+    value: str
+    important: bool
+
+
+class IssueOccurrenceData(TypedDict):
+    id: str
+    fingerprint: Sequence[str]
+    issue_title: str
+    subtitle: str
+    resource_id: str | None
+    evidence_data: Mapping[str, Any]
+    evidence_display: Sequence[IssueEvidenceData]
+    type: int  # Not sure if the consumer should convert to the appropriate type here as part of validation
+    detection_time: float  # Not sure if the consumer should convert to the appropriate type here as part of validation
+    event_id: Optional[str]
+    event: Optional[Event]
```

Comment:
Not sure if this will be an actual event or some other format that acts as a precursor to that